src/index.ts

```diff
@@ -1,47 +1,54 @@
 import ejs from "ejs";
-import { getContext, helpers, Property } from "@dylibso/xtp-bindgen";
+import { helpers, getContext, ObjectType, EnumType, ArrayType, XtpNormalizedType, MapType, Parameter, Property } from "@dylibso/xtp-bindgen"
 
-function toGolangType(property: Property): string {
-  if (property.$ref) return goName(property.$ref.name);
-  switch (property.type) {
-    case "string":
-      if (property.format === "date-time") {
-        return "time.Time";
-      }
-      return "string";
-    case "number":
-      if (property.format === "float") {
-        return "float32";
-      }
-      if (property.format === "double") {
-        return "float64";
+function toGolangTypeX(type: XtpNormalizedType, required?: boolean): string {
+  // annotate with null if nullable 
+  const pointerify = (t: string) => {
+    // TODO key off the `type.kind` here instead
+    if (t.startsWith("[]") || t.startsWith("map[")) return t
+    return `${type.nullable || !required ? '*' : ''}${t}`
+  }
+
+  switch (type.kind) {
+    case 'string':
+      return pointerify('string')
+    case 'int32':
+      return pointerify('int32')
+    case 'float':
+      return pointerify('float32')
+    case 'double':
+      return pointerify('float64')
+    case 'byte':
+      return pointerify('byte')
+    case 'date-time':
+      return pointerify('time.Time')
+    case 'boolean':
+      return pointerify('bool')
+    case 'array':
+      const arrayType = type as ArrayType
+      return pointerify(`[]${toGolangTypeX(arrayType.elementType)}`)
```

Comment:
Should arrays and maps never be pointer types? maybe i just remove this pointerify wrapper on these instead?

Reply:
Yea I think that's fine to keep as-is too.